Databases/CreateMangoDatabase.py

```diff
@@ -0,0 +1,209 @@
+                return False
+
+            result = self.collection.delete_one({"TilePoolId": tile_pool_id})
+            if result.deleted_count > 0:
+                logger.info(
+                    f"Tile pool with TilePoolId '{tile_pool_id}' deleted successfully."
+                )
+                return True
+            return False
+        except Exception as e:
+            logger.error(f"Failed to delete tile pool: {str(e)}")
+            return False
+
+    def delete_tile_pool_by_owner(
+        self, owner: str, requester: str, is_admin: bool = False
+    ) -> bool:
+        """Delete all tile pools for a specific owner if the requester is the owner or an admin."""
+        if requester != owner and not is_admin:
+            logger.error(
+                "Only the owner or an admin can delete all tile pools for this owner."
+            )
+            return False
+
+        try:
+            result = self.collection.delete_many({"Owner": owner})
+            if result.deleted_count > 0:
+                logger.info(
+                    f"Deleted {result.deleted_count} tile pools for owner '{owner}'."
+                )
+                return True
+            else:
+                logger.info(f"No tile pools found for owner '{owner}'.")
+                return False
+        except Exception as e:
+            logger.error(f"Failed to delete tile pools for owner '{owner}': {str(e)}")
+            return False
+
+    def delete_specific_tiles(
+        self,
+        tile_pool_id: str,
+        tiles_to_delete: list[str],
+        requester: str,
+        is_admin: bool = False,
+    ) -> bool:
+        """Delete specific tiles from a tile pool if the requester is the owner or an admin."""
+        try:
+            tile_pool = self.collection.find_one({"TilePoolId": tile_pool_id})
+            if tile_pool is None:
+                logger.info(f"Tile pool with TilePoolId '{tile_pool_id}' not found.")
+                return False
+
+            if tile_pool["Owner"] != requester and not is_admin:
+                logger.error(
+                    "Only the owner or an admin can delete tiles from this tile pool."
+                )
+                return False
+
+            # Remove specified tiles
+            updated_tiles = [
+                tile for tile in tile_pool["Tiles"] if tile not in tiles_to_delete
+            ]
+            result = self.collection.update_one(
+                {"TilePoolId": tile_pool_id}, {"$set": {"Tiles": updated_tiles}}
+            )
+            if result.modified_count > 0:
+                logger.info(
+                    f"Tiles {tiles_to_delete} deleted from tile pool '{tile_pool_id}'."
+                )
+                return True
+            return False
+        except Exception as e:
+            logger.error(f"Failed to delete tiles from tile pool: {str(e)}")
+            return False
+
+    def save_tile_pool_to_json(self, tile_pool: dict, filename: str) -> None:
```

Comment:
this is misplaced, tile pools are in the rest of backend are an object. A json serializer already exists in a different part of the project.